src/librustc_metadata/decoder.rs

```diff
@@ -1365,6 +1366,30 @@ impl<'a, 'tcx> CrateMetadata {
         // This shouldn't borrow twice, but there is no way to downgrade RefMut to Ref.
         self.source_map_import_info.borrow()
     }
+
+    /// Get the `DepNodeIndex` corresponding this crate. The result of this
+    /// method is cached in the `dep_node_index` field.
+    pub(super) fn get_crate_dep_node_index(&self, tcx: TyCtxt<'tcx>) -> DepNodeIndex {
+        let mut dep_node_index = self.dep_node_index.load();
+
+        if dep_node_index == DepNodeIndex::INVALID {
```

Comment:
Should we consider marking this `unlikely`/`cold`?

Reply:
Yeah, I can do that. (Though I have yet to see an instance where this actually makes a runtime difference `;)`)